src/libstd/io/stdio.rs

```diff
@@ -165,6 +165,9 @@ impl Stdin {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl Read for Stdin {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        // Flush stdout so that weird issues like a print!'d prompt not being
+        // shown until after the user hits enter.
+        drop(stdout().flush());
```

Comment:
Minor nit: `let _ = stdout().flush()` is a bit more common than `drop(stdout().flush())`.